src/main/java/pl/futurecollars/invoicing/db/file/FileBasedDatabase.java

```diff
@@ -0,0 +1,100 @@
+package pl.futurecollars.invoicing.db.file;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import lombok.AllArgsConstructor;
+import pl.futurecollars.invoicing.db.Database;
+import pl.futurecollars.invoicing.model.Invoice;
+import pl.futurecollars.invoicing.utils.FilesService;
+import pl.futurecollars.invoicing.utils.JsonService;
+
+@AllArgsConstructor
+public class FileBasedDatabase implements Database {
+
+  private final Path databasePath;
+  private final IdProvider idProvider;
+  private final FilesService filesService;
+  private final JsonService jsonService;
+
+  @Override
+  public int save(Invoice invoice) {
+    try {
+      invoice.setId(idProvider.getNextIdAndIncrement());
+      filesService.appendLineToFile(databasePath, jsonService.toJson(invoice));
+
+      return invoice.getId();
+    } catch (IOException ex) {
+      throw new RuntimeException("Database failed to save invoice", ex);
+    }
+  }
+
+  @Override
+  public Optional<Invoice> getById(int id) {
+    try {
+      return filesService.readAllLines(databasePath)
+          .stream()
+          .filter(line -> containsId(line, id))
+          .map(line -> jsonService.toObject(line, Invoice.class))
+          .findFirst();
+    } catch (IOException ex) {
+      throw new RuntimeException("Database failed to get invoice with id: " + id, ex);
+    }
+  }
+
+  @Override
+  public List<Invoice> getAll() {
+    try {
+      return filesService.readAllLines(databasePath)
+          .stream()
+          .map(line -> jsonService.toObject(line, Invoice.class))
+          .collect(Collectors.toList());
+    } catch (IOException ex) {
+      throw new RuntimeException("Failed to read invoices from file", ex);
+    }
+  }
+
+  @Override
+  public void update(int id, Invoice updatedInvoice) {
+    try {
+      List<String> allInvoices = filesService.readAllLines(databasePath);
+      var listWithoutInvoiceWithGivenId = allInvoices
```

Comment:
Zamiast wyszukiwać wszystkich faktur, które mają inny id, znajdź tylko ten, który chcesz update'ować. Dzięki temu nie będziesz miał dwóch list (które mogą być długie), lecz jedną listę i jeden obiekt. 

Po tej zmianie będziesz musiał również poprawić kod, który znajduje się w dalszej części metody.

Generalnie idea jest jedna: znajdź obiekt, który chcesz zaktualizować, usuń go z listy wszystkich invoice'ów, zaktualizuj go, dodaj zaktualizowany obiekt ponownie do listy, zapisz listę do pliku.

Reply:
List<String> invoices = filesService.readAllLines(databasePath);
      var oldInvoice = invoices.get(id);
      invoices.remove(oldInvoice);
      updatedInvoice.setId(id);
      invoices.add(jsonService.toJson(updatedInvoice));
      filesService.writeLinesToFile(databasePath, invoices);

Reply:
Poprawiłem, jakbyś mógł spojrzeć to będę wdzięczny